gregBorchers/Homework2GregBorchers-probs1-4NoCards.js

```diff
@@ -0,0 +1,359 @@
+// continue programming without it.  You decide to simulate the && operator with a
+// function!
+
+
+
+// **a)** Write a function `and2(a,b)` which tries to simulate the && operator: it
+// should always return the same result as `(a && b)` for any values of _a_ and
+// _b_.  (For example, `and2((0>1),true)` should return _false_.)  But you can't
+// use && itself within your function!
+
+var and2 = function(a,b){
+  return (a*b ? true : false); 
+};
+
+
+// **b) Write another function `and3(a,b,c)` which tries to simulate a double-&&
+// **operator: it should always return the same result as `(a && b && c)` for any
+// **values of a,b,c.  (For example, `and3((1>0),(0>1),true)` should return
+// **_false_.)  As before, you're not allowed to use && itself. Remember that you
+// **can call functions from functions!
+
+var and3 = function(a,b,c){
+  return ( a*b*c ? true : false);
+};
+
+
+// **c)** Now generalize your function to handle any number of values.  You will
+// learn better ways eventually, but for now use an array to store all the values.
+// Your new function `andN(values)` should accept one parameter _values_, which is
+// an array holding all the values to && together. If the argument you provide when
+// calling the function holds values [a,b,c...z], as in `andN(26,[a,b,c...z])`, the
+// function should return the same result as `(a && b && c && ... z)`. Make sure to
+// handle two special cases: length 0 (then return _true_) and length 1 (then
+// return that single value). Again, don't use &&.
+
+var andN = function(n, arr) {
+  
+  var retVal = false; // holds the return value
+  
+  if (arr[0] === null){
+    retVal = true;
+  }
+
+  retVal = and2(arr[0], arr[0]); // works for length 1
+  
+  for (i=1; i<arr.length; i++){
+    retVal = and2(retVal, arr[i]);
+  }
+  return (retVal ? true : false );  // && operator should return a boolean (and2 can return a number on certain args)
+};
+
+// testing section
+var testArray = [0,1,2,3,4,5,6,7,8,9];
+var returnedVal = andN(testArray.length, testArray)
+console.log("returned ", returnedVal);
+console.log("from this array: ", testArray);
+// returns false
+
+var testArray = [99,1,2,3,4,5,6,7,8,9];
+var returnedVal = andN(testArray.length, testArray)
+console.log("returned ", returnedVal);
+console.log("from this array: ", testArray);
+//returns true
+
+var testArray = [99,1,2,3,4,5,6,7,8,Infinity];
+var returnedVal = andN(testArray.length, testArray)
+console.log("returned ", returnedVal);
+console.log("from this array: ", testArray);
+//returns true
+
+var testArray = [2,1,2,3,4,5,6,7,8,NaN];
+var returnedVal = andN(testArray.length, testArray)
+console.log("returned ", returnedVal);
+console.log("from this array: ", testArray);
+//returns false
+
+// **d)** You've just realized that your effort was doomed: neither of your
+// functions can replace the && operator in certain circumstances.  Explain why,
+// and find an example which demonstrates failure.
+
+NaN or strings?
```

Comment:
There are ways around the issues of `NaN` or strings, but there are more basic problems that you can't get around. In particular, assume that you have an undeclared variable q. Compare `false&&q` with `and2(false, q)`.